### PR TITLE
[Wasm] Support for FileSavePicker

### PR DESCRIPTION
**Pull request by alextrepa**
GitHub Issue (If applicable): [#3215](https://github.com/unoplatform/uno/issues/3215) (Only adds support for wasm)

## PR Type
- Feature

## What is the current behavior?
Currently, the file save picker doesn't work on a WASM build. To save a file, you must do it manually.

## What is the new behavior?
The file save picker will now create a file in the browser cache (indexeddb). Then, once the cache manager is notified that the work is done, the browser will ask the user to download the file.

## PR Checklist
Please check if your PR fulfills the following requirements:

- [ ] Docs have been added/updated which fit [documentation template](https://github.com/unoplatform/uno/blob/master/doc/.feature-template.md) (for bug fixes / features)
- [ ] [Unit Tests and/or UI Tests](https://github.com/unoplatform/uno/blob/master/doc/articles/uno-development/working-with-the-samples-apps.md) for the changes have been added (for bug fixes / features) (if applicable)
- [ ] Validated PR `Screenshots Compare Test Run` results.
- [x] Contains **NO** breaking changes
- [x] Associated with an issue (GitHub or internal) ~and uses the [automatic close keywords]~(https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue).
- [x] Commits must be following the [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/#summary) specification.

### CODE REVIEW

File: src/Uno.UI/ts/Windows/Storage/FileSavePicker.ts [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+			}
+
+			const a = window.document.createElement('a');
+			const blob = new Blob([buffer]);
+
+			a.href = window.URL.createObjectURL(blob);
+			a.download = fileName;
+
+			document.body.appendChild(a);
+			a.click();
```
Comment (davidjohnoliver) [inline src/Uno.UI/ts/Windows/Storage/FileSavePicker.ts:19]: There's no more direct way of programmatically triggering a download?

File: src/Uno.UWP/Storage/Pickers/FileSavePicker.wasm.cs [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+		{
+			SuggestedStartLocation = PickerLocationId.DocumentsLibrary;
+			FileTypeChoices = new Dictionary<string, IList<string>>();
+		}
+
+		private async Task<StorageFile> PickFilesTask()
+		{
+			if (!FileTypeChoices.Any())
+			{
+				throw new COMException();
```
Comment (davidjohnoliver) [inline src/Uno.UWP/Storage/Pickers/FileSavePicker.wasm.cs:32]: Even if Windows throws a `COMException`, we usually go with a more meaningful exception type, `InvalidOperationException` would be valid here.
```suggestion
				throw new InvalidOperationException();
```

File: src/Uno.UI/ts/Windows/Storage/FileSavePicker.ts (hunk 1 above)
Comment (jeromelaban) [reply src/Uno.UI/ts/Windows/Storage/FileSavePicker.ts:19]: no... :)

File: src/Uno.UI/ActivityResultArgs.Android.cs [hunk 3]
```diff
@@ -9,9 +9,9 @@ namespace Uno.UI
 	/// <param name="ct">CancellationToken</param>
 	/// <param name="intent">The Intent you want to send</param>
 	/// <param name="requestCode">A specific Response code, this is useful if you send more than one type of request in parallel</param>
-	public class OnActivityResultArgs
+	internal class ActivityResultArgs
```
Comment (jeromelaban) [inline src/Uno.UI/ActivityResultArgs.Android.cs:12]: Should be in the other PR

File: src/Uno.UI/DelegateActivity.Android.cs [hunk 4]
```diff
@@ -24,13 +24,13 @@ namespace Uno.UI
         // Common example: Capture an image in landscape from an app locked to portrait (using MediaPickerService).
         ConfigurationChanges = ConfigChanges.Orientation | ConfigChanges.ScreenSize
     )]
-	public class DelegateActivity : BaseActivity
+	internal class DelegateActivity : BaseActivity
```
Comment (jeromelaban) [inline src/Uno.UI/DelegateActivity.Android.cs:27]: Should be in the other PR

File: src/Uno.UI/ts/Windows/Storage/FileSavePicker.ts (hunk 1 above)
Comment (jeromelaban) [reply src/Uno.UI/ts/Windows/Storage/FileSavePicker.ts:19]: no... :)

**Comment by jeromelaban** [comment]
Also, you'll need to squash your history to a single `feat:` commit

File: src/Uno.UI/ts/Windows/Storage/FileSavePicker.ts (hunk 1 above)
Comment (alextrepa) [reply src/Uno.UI/ts/Windows/Storage/FileSavePicker.ts:19]: for something as common as a simple download, you would think there would be a more elegant way. But that's seems to be the way to do it.

**Pull request merged**